tests/test_template_types.py

```diff
@@ -829,6 +829,26 @@ def test_letter_preview_renders_QR_code_correctly(jinja_template):
     assert jinja_template_locals["message"] == expected_qr_code_svg
 
 
+@mock.patch("notifications_utils.template.LetterPreviewTemplate.jinja_template.render")
+def test_letter_preview_url_with_query_params_renders_QR_code_correctly(jinja_template):
+    str(
+        LetterPreviewTemplate(
+            {
+                "content": "This is your link:\n\nqr: www.gov.uk/search?q=foo&r=bar",
```

Comment:
Think this needs parametrizing with a bunch of different content to make sure various characters are all being escaped/unescaped properly